test/test_sockets.py

```diff
@@ -350,6 +350,11 @@ def test_posix_proxy_sockets(self):
         # Build and run the TCP echo client program with Emscripten
         self.btest_exit('websocket/tcp_echo_client.c', args=['-lwebsocket', '-sPROXY_POSIX_SOCKETS', '-pthread', '-sPROXY_TO_PTHREAD'])
 
+  # Test that multiple pthreads calling send() on the same socket produces correct ordering semantics.
+  def test_sockets_send_immediately_after_connect(self):
+    with NodeJsWebSocketEchoServerProcess():
+      self.btest('sockets/send_immediately_after_connect.cpp', args=['-DSOCKET_DEBUG'], expected='0')
```

Comment:
Can you give the filename the full name of the test?  e.g. `test_sockets_send_immediately_after_connect.c`?   

Even though the file lives in the sockets subdirectory I find it very useful to have filenames and test names align, and we have been moving in that direction as much as possible (e.g. see the other other tests in this directory).